nin/frontend/app/scripts/controllers/bottom.js

```diff
@@ -184,10 +184,14 @@
         $scope.loopEnd = $scope.loopStart + newLoopLength;
       });
 
+      var lastFrame = 0;
       $scope.$watch('currentFrame', function (nextFrame) {
-        if ($scope.loopActive && nextFrame >= $scope.loopEnd) {
+        if ($scope.loopActive && !$scope.demo.music.paused && lastFrame >= $scope.loopStart 
+                              && lastFrame <= $scope.loopEnd && nextFrame >= $scope.loopEnd 
+                              && !(nextFrame >= $scope.loopEnd + 5)) {
```

Comment:
Consider setting lastFrame = nextFrame somehow when demo.jumpToFrame is called -- that way you won't have to use this somewhat arbitrary guard to allow manually jumping out of loops.